0.7.0-alpha.1 fails to build: `Undefined symbols for architecture ppc: "_getsectiondata", referenced from: _get_embedded_avm`

```
[913/1000] : && /opt/local/bin/gcc-mp-14 -pipe -Os -DNDEBUG -isystem/opt/local/include/LegacySupport -I/opt/local/include -arch ppc -mmacosx-version-min=10.6 -Wl,-search_paths_first -Wl,-headerpad_max_install_names -Wl,-headerpad_max_install_names -L/opt/local/lib -lMacportsLegacySupport src/platforms/generic_unix/CMakeFiles/AtomVM.dir/main.c.o -o src/AtomVM  -Wl,-rpath,/opt/local/lib  src/platforms/generic_unix/libAtomVM/liblibAtomVM.a  src/platforms/generic_unix/lib/liblibAtomVMDarwin-powerpc.a  src/platforms/generic_unix/libAtomVM/liblibAtomVM.a  src/platforms/generic_unix/lib/liblibAtomVMDarwin-powerpc.a  -lm  /opt/local/lib/libz.dylib  /opt/local/lib/libmbedtls.3.6.6.dylib  /opt/local/lib/libmbedx509.3.6.6.dylib  /opt/local/lib/libmbedcrypto.3.6.6.dylib  /opt/local/lib/libeverest.a  /opt/local/lib/libp256m.a && :
FAILED: [code=1] src/AtomVM 
: && /opt/local/bin/gcc-mp-14 -pipe -Os -DNDEBUG -isystem/opt/local/include/LegacySupport -I/opt/local/include -arch ppc -mmacosx-version-min=10.6 -Wl,-search_paths_first -Wl,-headerpad_max_install_names -Wl,-headerpad_max_install_names -L/opt/local/lib -lMacportsLegacySupport src/platforms/generic_unix/CMakeFiles/AtomVM.dir/main.c.o -o src/AtomVM  -Wl,-rpath,/opt/local/lib  src/platforms/generic_unix/libAtomVM/liblibAtomVM.a  src/platforms/generic_unix/lib/liblibAtomVMDarwin-powerpc.a  src/platforms/generic_unix/libAtomVM/liblibAtomVM.a  src/platforms/generic_unix/lib/liblibAtomVMDarwin-powerpc.a  -lm  /opt/local/lib/libz.dylib  /opt/local/lib/libmbedtls.3.6.6.dylib  /opt/local/lib/libmbedx509.3.6.6.dylib  /opt/local/lib/libmbedcrypto.3.6.6.dylib  /opt/local/lib/libeverest.a  /opt/local/lib/libp256m.a && :
Undefined symbols for architecture ppc:
  "_getsectiondata", referenced from:
      _get_embedded_avm in main.c.o
ld: symbol(s) not found for architecture ppc
collect2: error: ld returned 1 exit status
[914/1000] cd /opt/local/var/macports/build/atomvm-43bfe78f/work/build/examples/elixir/stm32 && elixirc /opt/local/var/macports/build/atomvm-43bfe78f/work/atomvm-0.7.0-alpha.1/examples/elixir/stm32/MultiBlink.ex
ninja: build stopped: subcommand failed.
```

Per AMP code:

The breakage is in `src/platforms/generic_unix/main.c`: the `__APPLE__` branch of `get_embedded_avm()` unconditionally calls `getsectiondata()` from `<mach-o/getsect.h>`. That symbol is only available in the modern macOS SDK / libSystem; it is not present on the legacy Mac OS X 10.5/10.6 PowerPC SDK that MacPorts uses for `-arch ppc` (`-mmacosx-version-min=10.6`), hence the `Undefined symbols for architecture ppc: "_getsectiondata"` link error.

Suggested minimal fix — fall through to the existing no-embedded-AVM `#else` branch on platforms where `getsectiondata` isn't available (same path FreeBSD already takes):

```diff
--- a/src/platforms/generic_unix/main.c
+++ b/src/platforms/generic_unix/main.c
@@
-#ifdef __APPLE__
+#include <Availability.h>
+
+#if defined(__APPLE__) && defined(__MAC_OS_X_VERSION_MIN_REQUIRED) \
+    && __MAC_OS_X_VERSION_MIN_REQUIRED >= 1070
+#define ATOMVM_APPLE_HAS_GETSECTIONDATA 1
+#endif
+
+#ifdef ATOMVM_APPLE_HAS_GETSECTIONDATA
 #include <mach-o/getsect.h>
 #include <mach-o/ldsyms.h>
 #elif defined(__linux__)
@@
 bool get_embedded_avm(const void **data, size_t *size)
 {
-#ifdef __APPLE__
+#ifdef ATOMVM_APPLE_HAS_GETSECTIONDATA
     // On macOS, look for the __ATOMVM,__avm_data section
     unsigned long section_size = 0;
     const void *section_data = getsectiondata(&_mh_execute_header, "__ATOMVM", "__avm_data", &section_size);
```

`getsectiondata()` was introduced in the 10.7 SDK, so gating the embedded-AVM support on `__MAC_OS_X_VERSION_MIN_REQUIRED >= 1070` lets PPC (and any other pre-10.7 deployment target) fall through to the existing `#else` branch — embedded AVM is simply unsupported there, matching the FreeBSD path. Alternatively the legacy `getsectdata()`/`getsectbyname()` APIs could be used for old PPC macOS, but the simple fallback is probably enough since embedded escriptize on legacy PPC is unlikely to be a real use case.

--- a/src/platforms/generic_unix/main.c
+++ b/src/platforms/generic_unix/main.c
@@ -27,9 +27,16 @@
 #include <unistd.h>
 
 #ifdef __APPLE__
+#include <Availability.h>
+// getsectiondata() was introduced in the macOS 10.7 SDK; on older deployment
+// targets (e.g. PowerPC builds targeting 10.6) embedded AVM is unsupported.
+#if defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 1070
+#define ATOMVM_HAS_EMBEDDED_AVM 1
 #include <mach-o/getsect.h>
 #include <mach-o/ldsyms.h>
+#endif
 #elif defined(__linux__)
+#define ATOMVM_HAS_EMBEDDED_AVM 1
 #include <errno.h>
 #include <fcntl.h>
 #include <sys/mman.h>
@@ -50,7 +57,7 @@
 #include "term.h"
 #include "utils.h"
 
-#ifdef __linux__
+#if defined(ATOMVM_HAS_EMBEDDED_AVM) && defined(__linux__)
 // On Linux (using ELF), the embedded avm binary is added as a section
 // and the following symbols are patched.
 uint64_t __atomvm_avm_offset __attribute__((section(".atomvm_avm_info")));
@@ -102,7 +109,7 @@ void print_help(const char *program_name)
  */
 bool get_embedded_avm(const void **data, size_t *size)
 {
-#ifdef __APPLE__
+#if defined(ATOMVM_HAS_EMBEDDED_AVM) && defined(__APPLE__)
     // On macOS, look for the __ATOMVM,__avm_data section
     unsigned long section_size = 0;
     const void *section_data = getsectiondata(&_mh_execute_header, "__ATOMVM", "__avm_data", &section_size);
@@ -112,7 +119,7 @@ bool get_embedded_avm(const void **data, size_t *size)
         *size = (size_t) section_size;
         return true;
     }
-#elif defined(__linux__)
+#elif defined(ATOMVM_HAS_EMBEDDED_AVM) && defined(__linux__)
     // On Linux, check if symbols were added by objcopy during escriptize
     if (__atomvm_avm_offset != 0 && __atomvm_avm_length != 0) {
         int fd = open("/proc/self/exe", O_RDONLY);
@@ -140,7 +147,8 @@ bool get_embedded_avm(const void **data, size_t *size)
         return true;
     }
 #else
-    // embedded avm not supported yet on this target (e.g. FreeBSD)
+    // embedded avm not supported yet on this target (e.g. FreeBSD,
+    // pre-10.7 macOS deployment targets such as PowerPC)
     UNUSED(data);
     UNUSED(size);
 #endif

